journal.py

```diff
@@ -41,48 +46,45 @@ class Entry(Base):
     created = sa.Column(sa.DateTime, nullable=False,
                         default=datetime.datetime.utcnow)
 
-    @property
-    def markdown(self):
-        output = ""
-        text = self.text.split("```")
-        for i in range(len(text)):
-            if i % 2 == 0:
-                output += markdown.markdown(text[i])
-            else:
-                output += highlight(text[i], PythonLexer(), HtmlFormatter())
-        return output
-
     @classmethod
-    def write(cls, title=None, text=None, session=None):
+    def all(cls, session=None):
         if session is None:
             session = DBSession
-        instance = cls(title=title, text=text)
-        session.add(instance)
-        return instance
+        return session.query(cls).order_by(cls.created.desc()).all()
 
     @classmethod
     def edit(cls, title=None, text=None, session=None, id=None):
         if session is None:
             session = DBSession
         instance = cls(title=title, text=text, id=id)
-        if title is not "" and text is not "":
-            session.query(cls).filter(cls.id == id).update(
-                {"title": title, "text": text})
-        else:
-            session.query(cls).filter(cls.id == id).delete()
+        session.query(cls).filter(cls.id == id).update(
+            {"title": title, "text": text})
```

Comment:
I love that you found the `update` method!  Good work!.  Given that you are using this, though, the line of code above at L59 is not really required, is it? Can you think of another way you might return the instance you've edited without needing to issue two separate DB requests?